graylog2-web-interface/src/views/components/searchbar/CSVExport.jsx

```diff
@@ -0,0 +1,154 @@
+// @flow strict
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import URI from 'urijs';
+import connect from 'stores/connect';
+/* $FlowFixMe: Need to add to flow typed */
+import styled from 'styled-components';
+
+import { StreamsStore } from 'views/stores/StreamsStore';
+import { FieldTypesStore } from 'views/stores/FieldTypesStore';
+import { ViewStore } from 'views/stores/ViewStore';
+import URLUtils from 'util/URLUtils';
+import { Modal, Button, Row } from 'components/graylog';
+import { Icon } from 'components/common';
+import ApiRoutes from 'routing/ApiRoutes';
+import StoreProvider from 'injection/StoreProvider';
+import View from 'views/logic/views/View';
+import Select from 'views/components/Select';
+
+const SessionStore = StoreProvider.getStore('Session');
+
+type Option = {
+  label: String,
+  value: String,
+};
+
+type Props = {
+  availableStreams: Array<Option>,
+  availableFields: Array<Option>,
+  closeModal: () => void,
+};
+
+const getURLForExportAsCSV = (selectedStream: ?string, selectedFields: Array<string> = []) => {
+  const { view } = ViewStore.getInitialState() || {};
+  if (view && view.type === View.Type.Search) {
+    const { queries } = view.search;
+    if (queries.size !== 1) {
+      throw new Error('Searches must only have a single query!');
+    }
+    const firstQuery = queries.first();
+    if (firstQuery) {
+      const { query: { query_string: queryString }, timerange } = firstQuery;
+      const query = !queryString ? '*' : queryString;
+      const url = new URI(URLUtils.qualifyUrl(
+        ApiRoutes.UniversalSearchApiController.export(
+          timerange.type,
+          query,
+          timerange,
+          selectedStream,
+          0,
+          0,
+          selectedFields,
+        ).url,
+      ));
+      if (URLUtils.areCredentialsInURLSupported()) {
+        url
+          .username(SessionStore.getSessionId())
+          .password('session');
+      }
+
+      return url.toString();
+    }
+  }
+  return new URI(URLUtils.qualifyUrl('/notfound'));
+};
+
+const wrapOption = o => ({ label: o, value: o });
+const defaultFields = ['timestamp', 'source', 'message'];
+const defaultFieldOptions = defaultFields.map(wrapOption);
+
+const CSVExport = ({ closeModal, availableStreams, availableFields }: Props) => {
```

Comment:
Sorting `availableStreams` and `availableFields` would really improve the usability.

Reply:
:+1: